File: src/main/java/com/kairlec/pojo/Json/FileInfo.java
package com.kairlec.pojo.Json;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.annotation.JSONField;
import com.alibaba.fastjson.serializer.SerializerFeature;
import lombok.Data;


@Data
public class FileInfo {
    private String name;
    private String type;
    private long size = 0;
    private long editTime = 0;
    private String description;
    private String path;
    @JSONField(serialize = false)
    private boolean exist;

    @Override
    public String toString() {
        return JSON.toJSONString(this, SerializerFeature.WriteMapNullValue);
    }

}

File: src/main/java/com/kairlec/pojo/json/FileInfo.kt
package com.kairlec.pojo.json;


data class FileInfo(
        var name: String,
        var type: String? = null,
        var size: Long = 0,
        var editTime: Long = 0,
        var path: String
) : Comparable<FileInfo> {
    override fun compareTo(other: FileInfo): Int {
        if (this.type == null || other.type == null) {
            return this.name.compareTo(other.name)
        }
        if (this.type == "folder" && other.type != "folder") {
            return -1
        }
        if (other.type == "folder" && this.type != "folder") {
            return 1
        }
        return this.name.compareTo(other.name)
    }

}
